shell/platform/windows/public/flutter_windows.h

```diff
@@ -57,6 +57,15 @@ typedef struct {
   // Array of Dart entrypoint arguments. This is deep copied during the call
   // to FlutterDesktopEngineCreate.
   const char** dart_entrypoint_argv;
+
+#ifdef WINUWP
+  // Number of elements in the array passed in as engine_switch_count.
+  int engine_switch_count;
+
+  // Array of Flutter engine switches. This is deep copied during the call
+  // to FlutterDesktopEngineCreate.
+  const char** engine_switches;
```

Comment:
This is problematic; these switches are not intended to be used as a compile-time API by developers, which is why the ability to set them programatically was removed from the desktop platforms. The concern here is that these flags do *not* have the same forward/backward compatibility guarantees as the rest of the API, so people shouldn't be encouraged (structurally) to bake them in. Environment variables are used now on the other desktop platforms specifically to indicate that they are transient.

I don't suppose there's any way to read the activation state directly from the engine via a global?

Reply:
OK GTK.  I don't believe so.  Another option would be to add an argument to `FlutterDesktopViewControllerCreateFromCoreWindow` accepting `Windows::ApplicationModel::Activation::LaunchActivatedEventArgs` and have the engine go from there.  If this works i'll refactor the PR else let's keep discussing options.

Reply:
Any thoughts here?

Reply:
That seems reasonable to me, given that UWP objects are already being passed through that API layer. The downside would be that it would make it more awkward to create view controllers from anywhere other than main, but I don't have a better suggestion there.

Reply:
If those objects are optional we could support that scenario too.  Ok I'll get it coded up.

Reply:
I've reworked the PR.